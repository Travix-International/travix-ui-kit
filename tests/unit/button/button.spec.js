import { shallow } from 'enzyme';
import React from 'react';
import Button from '../../../components/button/button';

describe('Button', () => {
  describe('#render()', () => {
    const onClickSpy = () => {};

    it('should render disabled link', () => {
      const wrapper = shallow(
        <Button disabled href="http://google.com" size="xs" type="link">Extra small</Button>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods #1', () => {
      const wrapper = shallow(
        <Button disabled size="xs" type="link">Extra small</Button>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should not modify mods', () => {
      const mods = ['test'];
      const wrapper = shallow(
        <Button
          mods={mods}
          onClick={onClickSpy}
        >
          Button
        </Button>
      );

      expect(mods.length).toEqual(1);
      expect(mods[0]).toEqual('test');
      expect(wrapper).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods #2', () => {
      const wrapper = shallow(
        <Button dataAttrs={{ gtm: 'id' }} size="xs" type="submit">Extra small</Button>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods #3', () => {
      const wrapper = shallow(
        <Button onClick={onClickSpy} size="xs" type="button">Extra small</Button>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render gtm data attribute', () => {
      const wrapper = shallow(
        <Button dataAttrs={{ gtm: 'id' }} onClick={onClickSpy}>Extra small</Button>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods without data-attrs', () => {
      const wrapper = shallow(
        <Button onClick={onClickSpy} type="button">Extra small</Button>
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render ghost variation', () => {
      const wrapper = shallow(
        <Button onClick={onClickSpy} type="button" variation="ghost">Ghost</Button>
      );

      expect(wrapper.find('.ui-button').hasClass('ui-button_variation_ghost')).toEqual(true);
    });
  });
});
