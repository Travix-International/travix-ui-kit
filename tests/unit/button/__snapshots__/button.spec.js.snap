// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button #render() should not modify mods 1`] = `
<button
  className="ui-button ui-button_test ui-button_size_m ui-button_variation_default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Button
</button>
`;

exports[`Button #render() should render disabled link 1`] = `
<a
  className="ui-button ui-button_size_xs ui-button_variation_default ui-button_disabled_true"
  href="http://google.com"
>
  Extra small
</a>
`;

exports[`Button #render() should render gtm data attribute 1`] = `
<button
  className="ui-button ui-button_size_m ui-button_variation_default"
  data-gtm="id"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Extra small
</button>
`;

exports[`Button #render() should return base class with mods for strings as class and string as mods #1 1`] = `""`;

exports[`Button #render() should return base class with mods for strings as class and string as mods #2 1`] = `
<button
  className="ui-button ui-button_size_xs ui-button_variation_default"
  data-gtm="id"
  disabled={false}
  type="submit"
>
  Extra small
</button>
`;

exports[`Button #render() should return base class with mods for strings as class and string as mods #3 1`] = `
<button
  className="ui-button ui-button_size_xs ui-button_variation_default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Extra small
</button>
`;

exports[`Button #render() should return base class with mods for strings as class and string as mods without data-attrs 1`] = `
<button
  className="ui-button ui-button_size_m ui-button_variation_default"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Extra small
</button>
`;
